src/rez/package_cache.py

```diff
@@ -214,17 +229,46 @@ def add_variant(self, variant, force=False):
                     % package.repository
                 )
 
-        no_op_statuses = (
+        no_op_statuses = {
             self.VARIANT_FOUND,
-            self.VARIANT_COPYING,
-            self.VARIANT_COPY_STALLED
-        )
+            self.VARIANT_COPY_STALLED,
+        }
+        if not wait_for_copying:
+            # Copying variants are only no-ops if we want to ignore them.
+            no_op_statuses.add(self.VARIANT_COPYING)
 
         # variant already exists, or is being copied to cache by another thread/proc
         status, rootpath = self._get_cached_root(variant)
         if status in no_op_statuses:
+            if logger:
+                logger.warning(f"Not caching {variant.name}-{variant.version}. "
```

Comment:
I think we should use the qualified name which will include the variant id.
```suggestion
                logger.warning(f"Not caching {variant.qualified_name}. "
```
The same applies to all the other places where `{variant.name}-{variant.version}` is use.

Reply:
:+1: